lib/dartdoc.dart

```diff
@@ -53,43 +53,58 @@ class DartdocFileWriter implements FileWriter {
   DartdocFileWriter(this.outputDir, this.resourceProvider);
 
   @override
-  void write(String filePath, Object content,
-      {bool allowOverwrite, Warnable element}) {
+  void writeBytes(
+    String filePath,
+    List<int> content, {
+    bool allowOverwrite = false,
+  }) {
     // Replace '/' separators with proper separators for the platform.
     var outFile = path.joinAll(filePath.split('/'));
 
-    allowOverwrite ??= false;
     if (!allowOverwrite) {
-      if (_fileElementMap.containsKey(outFile)) {
-        assert(element != null,
-            'Attempted overwrite of $outFile without corresponding element');
-        var originalElement = _fileElementMap[outFile];
-        Iterable<Warnable> referredFrom =
-            originalElement != null ? [originalElement] : null;
-        element?.warn(PackageWarning.duplicateFile,
-            message: outFile, referredFrom: referredFrom);
-      }
+      _warnAboutOverwrite(outFile, null);
     }
-    _fileElementMap[outFile] = element;
```

Comment:
i wonder if the test failures might be due to this being removed?

Reply:
Ah ha, good catch!